File: src/main/webapp/service/warehouseCardService.js
/**
 * Created by Olivera on 3.7.2017..
 */
app.service('warehouseCardService', function ($http) {
    return {
        readAll: function(onSuccess, onError){
            $http.get('/api/warehouseCards').then(onSuccess, onError);
        },
        read: function(warehouse, onSuccess, onError) {
            $http.get('/api/warehouseCards/' + warehouse.id).then(onSuccess, onError);
        },
        generateReport: function(data, onSuccess, onError){
            $http.post('/api/warehouseCards/generateReport', data).then(onSuccess, onError);
        },
        level: function (warehouseCard, onSuccess, onError) {
            $http.post('/api/warehouseCards/level', warehouseCard).then(onSuccess, onError);
        },
        getCardForWare: function (ware, onSuccess, onError) {
            $http.post('/api/warehouseCards/card', ware).then(onSuccess, onError);
        }
    }
});

File: src/main/java/com/ftn/service/WarehouseCardService.java
package com.ftn.service;

import com.ftn.model.dto.BusinessYearDTO;
import com.ftn.model.dto.WareDTO;
import com.ftn.model.dto.WarehouseCardDTO;
import com.ftn.model.dto.WarehouseDTO;
import com.ftn.model.dto.*;

import java.io.ByteArrayOutputStream;
import java.io.File;
import java.text.ParseException;
import java.util.Date;

import java.util.List;

/**
 * Created by JELENA on 31.5.2017.
 */
public interface WarehouseCardService {

    List<WarehouseCardDTO> read();

    //read warehouse cards for active warehouse
    List<WarehouseCardDTO> read(Long id);

    //Check if there's card for ware, year and specific warehouse
    WarehouseCardDTO read(WareDTO wareDTO, BusinessYearDTO businessYearDTO, WarehouseDTO warehouseDTO);

    WarehouseCardDTO create(WarehouseCardDTO warehouseCardDTO);

    WarehouseCardDTO update(Long id, WarehouseCardDTO warehouseCardDTO);

    void delete(Long id);

    String generateReport(ReportDataDTO reportDataDTO);

    WarehouseCardDTO getWarehouseCardForWare(WareDTO wareDTO);
}
